### PR TITLE
adding fixes to pagination, redirection to patient profile and no res…

### PR DESCRIPTION
**Pull request by hclarkEnq**
## Description

Rendering and displaying all the search results and adding the colors to the display.

 * Additionally fixing issues with redirection to patient profile 
 * Adding fix to pagination when there is no pages to display should not be pagination so added check there

## Tickets

* [CNFT1-2444](https://cdc-nbs.atlassian.net/browse/CNFT1-2444)

## Checklist before requesting a review
- [ ] PR focuses on a single story
- [ ] Code has been fully tested to meet acceptance criteria
- [ ] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [ ] All new functions/classes/components reasonably small
- [ ] Functions/classes/components focused on one responsibility
- [ ] Code easy to understand and modify (clarity over concise/clever)
- [ ] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [ ] PR does not contain hardcoded values (Uses constants)
- [ ] All code is covered by unit or feature tests

[CNFT1-2444]: https://cdc-nbs.atlassian.net/browse/CNFT1-2444?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/apps/search/layout/result/list/SearchResultList.tsx [hunk 1]
```diff
@@ -8,11 +8,18 @@ type Props<T> = {
...
+        results.map((result, index) => (
+            <div className={styles.result} key={index}>
+                {render(result)}
+            </div>
+        ))
+    ) : (
+        <div className="text-center">
+            <p>No results found.</p>
+            <p>Try refining your search.</p>
         </div>
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/layout/result/list/SearchResultList.tsx:21]: issue:  This is for story [CNFT1-2469](https://cdc-nbs.atlassian.net/browse/CNFT1-2469) that Alaap is working on.  The `SearchResults` component is meant to be displayed only if there are results to show.  The [SearchLayout](https://github.com/CDCgov/NEDSS-Modernization/blob/main/apps/modernization-ui/src/apps/search/layout/SearchLayout.tsx) is going to handle which component is displayed based on the out come of the search

File: apps/modernization-ui/src/apps/search/patient/result/list/PatientSearchResultListItem.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ const PatientSearchResultListItem = ({ result }: Props) => {
                     <Link
                         id="legalName"
                         className={`${styles.value}, ${styles.name}`}
-                        to={`patient-profile/${result.patient}/summary`}>
+                        to={`/patient-profile/${result.shortId}/summary`}>
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/patient/result/list/PatientSearchResultListItem.tsx:30]: praise:  Thanks for catching that

File: apps/modernization-ui/src/design-system/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -22,5 +22,7 @@ export const Pagination = () => {
                 />
             </div>
         </div>
+    ) : (
+        <></>
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/design-system/Pagination/Pagination.tsx:26]: question:  What's the reason for hiding this?  shouldn't the total pages always be there?

File: apps/modernization-ui/src/apps/search/useSearch.ts [hunk 4]
```diff
@@ -98,7 +98,7 @@ const useSearch = <C, A, R>({ transformer, resultResolver, termResolver }: Setti
                 direction: asSortDirection(direction)
             };
         }
-    }, [direction]);
+    }, [property, direction]);
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/useSearch.ts:101]: praise:  Thanks for catching that

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]
Please take out the changes that overlap with [CNFT1-2469](https://cdc-nbs.atlassian.net/browse/CNFT1-2469)

File: apps/modernization-ui/src/design-system/Pagination/Pagination.tsx (hunk 3 above)
Comment (hclarkEnq) [reply apps/modernization-ui/src/design-system/Pagination/Pagination.tsx:26]: I saw this when I was making a search and there were no results I was seeing infinite pages displaying. So thats why I added a condition on that total number of results

Comment (adamloup-enquizit) [reply apps/modernization-ui/src/design-system/Pagination/Pagination.tsx:26]: ahh ok, that should hopefully go away with the No results story

**Review by adamloup-enquizit — APPROVED** [review]
Tony is about to post a bug for that patient profile link issue

**Review by jlee-enquizit — APPROVED** [review]

**Pull request merged**